overlord/devicestate/devicemgr.go

```diff
@@ -1077,7 +1077,7 @@ func (m *DeviceManager) ensureTriedRecoverySystem() error {
 	if release.OnClassic {
 		return nil
 	}
-	if m.systemMode != "run" {
+	if m.SystemMode() != "run" {
```

Comment:
this means this run also on UC16/18 now, is that intended

Reply:
Dropped this bit and left a couple of comments about why we should use m.systemMode rather than the helper (at least until we can sort out the tests to do something more reasonable)